### ISSUE TITLE
pgbackrest backup and archive-get doesnt work together

### ISSUE THREAD

**Issue by patagh** [pgbackrest/pgbackrest#1620]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: 2.36

2. PostgreSQL version: 13

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each: I use patroni cluster with 2 nodes on RHEL8

4. Did you install pgBackRest from source or from a package? from package

5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

7. Describe the issue:

I have a patroni cluster with 2 nodes. I take pgbackrest backup from standby to NFS. I got an error when I started to full backup. The error is: unable to find primary cluster - cannot proceed
I know that error and I add pg1-host config into pgbackrest.conf on the standby node. It works now, however, my archive-get command is failing now. The archive-get command works when I remove the pg1-host config from standby node pgbackrest. conf, however, the backup process is failing now. How can I handle this problem? Can I guess me?
info: my restore command on standby is : pgbackrest --stanza=titanium archive-get %f %p

PRIMARY pgbackrest.conf:

[global]
repo1-host=titanium02
repo1-host-user=postgres
repo1-path=/nfsbackup/titanium
repo1-retention-full=6
process-max=48
log-level-file=detail
archive-async=y
[global:archive-push]
compress-level=6
process-max=16
archive-push-queue-max=10GB
[titanium]
pg1-host-user=postgres
pg1-path=/pgdata/data
pg1-port=5432
start-fast=y

STANDBY pgbackrest.conf:

[global]
repo1-path=/nfsbackup/titanium
repo1-host-user=postgres
repo1-retention-full=6
process-max=48
log-level-file=detail
archive-async=y
backup-standby=y
lock-path=/nfsbackup/titanium/lock
delta=y
[global:archive-push]
compress-level=6
process-max=16
archive-push-queue-max=10GB
[global:archive-get]
process-max=16
archive-get-queue-max=10GB
[titanium]
**#pg1-host=titanium01**
pg1-host-user=postgres
pg1-path=/pgdata/data
pg1-port=5432
recovery-option=type=standby
recovery-option=primary_conninfo=host=titanium01 user=replicator passfile=/pgdata/.pgpass port=5432 sslmode=prefer application_name=titanium02 gssencmode=prefer
recovery-option=recovery_target_timeline=latest
start-fast=y

**Comment by pgstef** [comment]
Hi,

First a all, a few remarks about your PRIMARY configuration:
* `repo1-host` and `repo1-path` should be exclusives (use one or the other). I mean, or you're using a dedicated repo/backup host, or you're pushing things to a shared path (nfs mount?).
* `pg1-host-user` should be `pg1-user` since it's reflecting the locally running cluster.
* if you don't take backups here, `repo1-retention-full` is useless.

On your STANDBY server, the **backup** and *archive-get** commands can work together. You have to define both PG clusters in your config. Something like this should be enough:

```ini
...
start-fast=y
[titanium]
pg1-host=titanium01
pg1-path=/pgdata/data
pg2-path=/pgdata/data
recovery-option=primary_conninfo=host=titanium01...
```

* I believe `recovery-option=type=standby` is invalid. You should use the `restore --type=standby` option.
* `recovery-option=recovery_target_timeline=latest` that one is default since PG 12

Kind Regards,

**Comment by patagh** [comment]
Hi Stefan,
Thank you so much for your explanation.
I use the NFS mount on the standby node to take backup or push archives. So, the repo is the NFS mount point on the standby node. Also, I added the pg1-host and pg2-host and pg2-host, it works fine, backup and archive-get commands work together. Thank you. Also, recovery-option=type=standby parameter is invalid that you said, yes you are right. I removed this parameter too.

Thank you so much again.
Best regards,
